src/main/java/com/hrznstudio/galacticraft/block/decoration/MoonCheeseBlock.java

```diff
@@ -33,19 +34,15 @@
 /**
  * @author <a href="https://github.com/StellarHorizons">StellarHorizons</a>
  */
-public class MoonCheeseBlock extends CakeBlock {
-
-    private static final IntProperty BITES = IntProperty.of("bites", 0, 6);
-    private static final BooleanProperty WATERLOGGED = Properties.WATERLOGGED;
-
+public class MoonCheeseBlock extends CakeBlock implements Waterloggable {
     public MoonCheeseBlock(Settings settings) {
         super(settings);
-        this.setDefaultState(this.getStateManager().getDefaultState().with(WATERLOGGED, false).with(BITES, 0));
+        this.setDefaultState(this.getStateManager().getDefaultState().with(Properties.WATERLOGGED, false));
     }
 
```

Comment:
```
    @Override
     public ActionResult onUse(BlockState state, World world, BlockPos pos, PlayerEntity player, Hand hand, BlockHitResult hit) {
        ItemStack itemStack = player.getStackInHand(hand);
        if (itemStack.getItem().equals(Items.WATER_BUCKET)) {
            world.setBlockState(pos, state.with(Properties.WATERLOGGED, true));
            player.inventory.setStack(player.inventory.selectedSlot, new ItemStack(Items.BUCKET));
            return ActionResult.SUCCESS;
        } else if (itemStack.getItem().equals(Items.BUCKET)) {
            world.setBlockState(pos, state.with(Properties.WATERLOGGED, false));
            player.inventory.setStack(player.inventory.selectedSlot, new ItemStack(Items.WATER_BUCKET));
            return ActionResult.SUCCESS;
        } else
            return super.onUse(state, world, pos, player, hand, hit);
    }
```
This might help a bit